src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -450,7 +450,9 @@ JRT_END
 
 // Reference implementation at src/java.base/share/classes/java/lang/Float.java:floatToFloat16
 JRT_LEAF(jshort, SharedRuntime::f2hf(jfloat  x))
-  jint doppel = SharedRuntime::f2i(x);
+  union {jfloat f; juint i;} bits;
+  bits.f = x;
+  jint doppel = bits.i;
```

Comment:
Doesn't the conversion from unsigned to signed risk a compiler warning being emitted?

Can't you just use the existing `JavaValue` type to perform the union conversion trick?

Reply:
Hi David, thanks for pointing this out. I have updated the code to use jint.
I have used the union conversion trick that was previously used in SharedRuntime::drem and SharedRuntime::frem. I hope that's okay with you.

Reply:
We seem to employ this trick in a few places, e.g. also see metaprogramming/primitiveConversions.hpp. It would be good to reduce that so I will file a separate RFE.